CMakeLists.txt

```diff
@@ -112,6 +112,20 @@ endif()
 add_definitions(-Wall -Wextra -pedantic -Wno-unused-parameter
     -Wstrict-prototypes -std=gnu99)
 
+include(CheckCCompilerFlag)
+check_c_compiler_flag(-fstack-protector-strong HAS_FSTACK_PROTECTOR_STRONG_FLAG)
+check_c_compiler_flag(-fstack-protector HAS_FSTACK_PROTECTOR_FLAG)
+
+if(HAS_FSTACK_PROTECTOR_STRONG_FLAG)
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector-strong")
+elseif(HAS_FSTACK_PROTECTOR_FLAG)
+  message(WARNING "-fstack-protector-strong is unsupported, falling back to -fstack-protector")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector --param ssp-buffer-size=4")
+else()
+  message(WARNING "-fstack-protector-strong and -fstack-protector are unsupported, disabling")
```

Comment:
Does `check_c_compiler_flag()` not already leave some indication of whether a check passed?